### PR TITLE
Remove enum variant expr

### PR DESCRIPTION
**Pull request by Havvy**
This PR builds on top of #962, so that one needs to be approved and merged before this one.

This PR also gets rid the enum variant expression page. I think it was originally considered different when the expressions page was split into multiple chapters, and then quietly ignored. The struct expression page gives the same amount of information, plus more. It could give more (e.g. mentioning that it sets the discriminant), but I'll do that when I clean up the struct expression page itself.

### CODE REVIEW

**Comment by ehuss** [comment]
Can you rebase so the diff is easier to read?

**Comment by Havvy** [comment]
Rebased.

File: src/expressions/struct-expr.md [hunk 1]
```diff
@@ -101,7 +101,8 @@ Point3d { x, y: y_value, z };
 ## Tuple struct expression
 
 A struct expression with fields enclosed in parentheses constructs a tuple struct.
-Though it is listed here as a specific expression for completeness, it is equivalent to a [call expression] to the tuple struct's constructor. For example:
+Though it is listed here as a specific expression for completeness, it is equivalent to a [call expression] to the tuple struct's constructor. For example: <!--
+This is false. Example: Position(0, ..other_position) -->
```
Comment (ehuss) [inline src/expressions/struct-expr.md:105]: I don't understand this comment. Can you show what it means? 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a295289c9b9e281e95c80e220b3fd3ad

**Review by ehuss — COMMENTED** [review]
Can you add this to `book.toml`:

```toml
[output.html.redirect]
"/expressions/enum-variant-expr.html" = "struct-expr.html"
```

File: src/expressions/struct-expr.md (hunk 1 above)
Comment (Havvy) [reply src/expressions/struct-expr.md:105]: Oh, I figured that would work. Guess it does not, and its falseness is false.

**Comment by ehuss** [comment]
Also, should there be a mention somewhere that the enum variant doesn't support functional record update syntax?  Or did I miss that?

**Comment by Havvy** [comment]
Added the redirect, removed the comment, and changed to enum variant.

Also added a thing about the syntax only working for struct type values. It's a bit wordy, and I'll do a cleanup pass on it later.

**Review by ehuss — APPROVED** [review]
Thanks!
